### DISCUSSION TITLE
Schema for Instrument object

### DISCUSSION THREAD

**Discussion by dselva80** [eose-tools-org/eose-api#9] [General]
Alex will add needs for spacecraft design algorithm. Mass, power, datarate etc.
Vinay will add needs for instruPy.

**Comment by Snaggerdoodle** [comment]
The current payload JSON input for the spacecraft design module looks like this:

"payload": {
        "mass": 2.0,
        "dimensions": [
            0.2,
            0.1,
            0.1
        ],
        "avgPower": 10.0,
        "peakPower": 10.0,
        "name": "sLDU345",
        "tempRange": [
            -10.0,
            50.0
        ],
        "resolution": 1.7936070108330142,
        "FOV": 2.1,
        "specRange": "Visible",
        "dataRate": 6.0
    }

**Comment by vinay0000** [comment]
I have added an initial Instruments model (focusing on the geometric and physical properties) and a corresponding Jupyter notebook with instantiation examples in the following locations (on a separate branch off of **dev-tatc-analysis**):

Branch: **dev-vinay**

[Instruments Model](https://github.com/eose-tools-org/eose-api/blob/dev-vinay/src/eose/instruments.py)
[Jupyter Notebook Examples](https://github.com/eose-tools-org/eose-api/blob/dev-vinay/src/instrument-example.ipynb)

There are additional advanced parameters for the BasicSensor model, which can be updated after reaching a consensus on the initial version. 
These parameters relate to specifications of maneuver, Field of Regard (FOR), scene-FOV, generic spherical polygon for FOV specification.

Following this step, we can add more models specific to particular types of instruments such as optical sensors, SARs, and radiometers.

**Comment by vinay0000** [comment]
A version of the instruments example involving TAT-C (for propagation and coverage) and InstruPy (for datametrics) is in the branch `dev-vinay`. 

It calculates the metrics for a "basic sensor" (range, look angle, incidence angle, solar zenith @ target point)

The relevant files are:

[Instruments schema](https://github.com/eose-tools-org/eose-api/blob/dev-vinay/src/eose/instruments.py)
[Data metrics schema](https://github.com/eose-tools-org/eose-api/blob/dev-vinay/src/eose/datametrics.py)
[Example](https://github.com/eose-tools-org/eose-api/blob/dev-vinay/src/instrument-example.ipynb)

**Comment by vinay0000** [comment]
Question: 

Should we can include a **maneuver** attribute to the instrument? Based on the maneuver and instrument FOV specifications, the FOR can be calculated. 

@ptgrogan

**Reply by ptgrogan** [reply]
That is a great suggestion! Maybe label it more descriptively, though, like `max_off_nadir_pointing` or similar?
